tests/integration/cli/test_cli.py

```diff
@@ -46,18 +45,18 @@ def test_otx_train(recipe: str, tmp_path: Path) -> None:
     task = recipe.split("/")[0]
     model_name = recipe.split("/")[1].split(".")[0]
     tmp_path = tmp_path / f"otx_train_{model_name}"
-    command_line = [
+    command_cfg = [
         "otx", "train",
         f"+recipe={recipe}",
         f"base.data_dir={DATASET[task]['data_dir']}",
-        "trainer.max_epochs=1",
-        "trainer.check_val_every_n_epoch=1",
-        "trainer=auto",
         f"base.work_dir={tmp_path}",
         f"base.output_dir={tmp_path / 'outputs'}",
+        "+debug=intg_test",
         *DATASET[task]['overrides'],
     ]
-    rc, stdout, _ = check_run(command_line)
+    command = " ".join(command_cfg)
+
+    rc = os.system(command=command)
```

Comment:
Can this execute the CLI process in the main process without spawning another subprocess? I mean, if I execute `pdb -m pytest tests/integration/cli/test_cli.py`, is it possible to break at the model training code? My suggestion was that developers should be easy to debug when there is problem on the intergration tests.

Reply:
https://github.com/openvinotoolkit/training_extensions/pull/2701/commits/a39b283ddfcb36314a320572d1b692166cc2eaa9
-> I followed your advice and modified it to call the `main` function directly, but had a problem in this case: I noticed that  if the module is registered in the registry in the first test, it affects the next test. (test case with first detection recipe is passed, but others are failed) So for now, I couldn't think of any other good way to solve this, so I changed the registry registration to force in `src/otx/core/model/entity/detection.py`, 

I'm sure there's a way to prevent this in tests only without touching existing code, but I can't think of it at the moment - if you know of one, please let me know.